app/src/pages/CreateProject.svelte

```diff
@@ -0,0 +1,364 @@
+<script>
+  import ModalLayout from "../layouts/ModalLayout.svelte";
+  import { showNotification } from "../stores.js";
+  import { slide } from "svelte/transition";
+
+  import { gql } from "apollo-boost";
+  import { getClient, query, mutate } from "svelte-apollo";
+
+  import * as path from "../path.js";
+
+  import { pop, push } from "svelte-spa-router";
+  import {
+    Button,
+    Header,
+    Icon,
+    Input,
+    DirectoryInput,
+    CheckboxInput,
+    Text,
+    Title,
+    Select
+  } from "../DesignSystem";
+
+  const options = {
+    NEW: "new",
+    EXISTING: "existing"
+  };
+
+  const DEFAULT_BRANCH = "master";
+
+  let currentSelection;
+
+  const selectNew = () => {
+    currentSelection = options.NEW;
+  };
+  const selectExisting = () => {
+    currentSelection = options.EXISTING;
+  };
+  $: isNew = currentSelection === options.NEW;
+  $: isExisting = currentSelection === options.EXISTING;
+
+  let name = "";
+  let description = "";
+  let defaultBranch = DEFAULT_BRANCH;
+  let publish = true;
+  let newRepositoryPath = "";
+  let existingRepositoryPath = "";
+  let imageUrl = "";
+
+  const VALID_NAME_MATCH = new RegExp("^[a-z0-9][a-z0-9_-]+$", "i");
+  let showValidations = false;
+
+  $: isFormValid = isNameValid && isPathValid;
+  $: isNameValid = VALID_NAME_MATCH.test(name);
+  $: isPathValid =
+    (isNew && newRepositoryPath.length > 0) ||
+    (isExisting && existingRepositoryPath.length > 0);
+
+  const client = getClient();
+
+  const LOCAL_BRANCHES = gql`
+    query($path: String!) {
+      localBranches(path: $path)
+    }
+  `;
+
+  const CREATE_PROJECT = gql`
+    mutation($metadata: MetadataInput!, $path: String!, $publish: Boolean!) {
+      createProject(metadata: $metadata, path: $path, publish: $publish) {
+        id
+        metadata {
+          name
+        }
+      }
+    }
+  `;
+
+  const anonymize = str => {
+    return str
+      .split("")
+      .reduce(
+        (prevHash, currVal) =>
+          ((prevHash << 5) - prevHash + currVal.charCodeAt(0)) | 0,
+        0
+      );
+  };
+
+  const createProject = async () => {
+    let response;
+
+    try {
+      response = await mutate(client, {
```

Comment:
Is this how we do manual gql queries? Pretty cool.

Reply:
Yep: https://github.com/timhall/svelte-apollo#mutate